File: desafioleds/src/main/java/io/github/jessejuniordev/desafioleds/model/Concurso.java
package io.github.jessejuniordev.desafioleds.model;

import jakarta.persistence.*;
import lombok.Data;

import java.util.UUID;

@Entity
@Table(name = "concursos")
@Data
public class Concurso {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(nullable = false)
    private UUID id;

    @Column(length = 100, nullable = false)
    private String orgao;

    @Column(length = 10, nullable = false)
    private String edital;

    @Column(name = "codigo_concurso", length = 20, nullable = false)
    private String codigoConcurso;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String vagas; // armazena dados com modelo: ["carpinteiro", "marceneiro"]
}

File: old/app/models/concurso.py
from sqlalchemy import Column, String
from sqlalchemy.dialects.postgresql import ARRAY
from ..database import Base

class Concurso(Base):
    __tablename__ = 'concursos'
    codigo_concurso = Column(String, primary_key=True, index=True)
    orgao = Column(String, index=True, nullable=False)
    edital = Column(String, nullable=False)
    vagas = Column(ARRAY(String))